superset/assets/src/dashboard/components/FilterIndicatorsContainer.jsx

```diff
@@ -0,0 +1,162 @@
+
+const propTypes = {
+  // from  props
+  dashboardFilters: PropTypes.object.isRequired,
+  chartId: PropTypes.number.isRequired,
+  chartStatus: PropTypes.string,
+
+  // from redux
+  filter_immune_slices: PropTypes.arrayOf(PropTypes.number).isRequired,
+  filter_immune_slice_fields: PropTypes.object.isRequired,
+  setDirectPathToChild: PropTypes.func.isRequired,
+};
+
+const defaultProps = {
+  chartStatus: 'loading',
+};
+
+export default class FilterIndicatorsContainer extends React.PureComponent {
+  getFilterIndicators() {
+    const {
+      dashboardFilters,
+      chartId: currentChartId,
+      filter_immune_slices,
+      filter_immune_slice_fields,
+    } = this.props;
+
+    if (Object.keys(dashboardFilters).length === 0) {
+      return [];
+    }
+
+    const dashboardFiltersColorMap = getFilterColorMap();
+
+    const sorteIndicatorsByEmptiness = Object.values(dashboardFilters).reduce(
+      (indicators, dashboardFilter) => {
+        const {
+          chartId,
+          componentId,
+          directPathToFilter,
+          scope,
+          isDateFilter,
+          isInstantFilter,
+          columns,
+        } = dashboardFilter;
+
+        // do not apply filter on filter_box itself
+        // do not apply filter on filter_immune_slices list
+        if (
+          currentChartId !== chartId &&
+          !filter_immune_slices.includes(currentChartId)
+        ) {
+          Object.keys(columns).forEach(name => {
+            const colorMapKey = getFilterColorKey(chartId, name);
+            const directPathToLabel = directPathToFilter.slice();
+            directPathToLabel.push(`LABEL-${name}`);
+            const indicator = {
+              chartId,
+              colorCode: dashboardFiltersColorMap[colorMapKey],
+              componentId,
+              directPathToFilter: directPathToLabel,
+              scope,
+              isDateFilter,
+              isInstantFilter,
+              name,
+              values: columns[name],
+            };
+
+            // do not apply filter on fields in the filter_immune_slice_fields map
+            if (
+              filter_immune_slice_fields[currentChartId] &&
+              filter_immune_slice_fields[currentChartId].includes(name)
+            ) {
+              return;
+            }
+
+            if (isEmptyValue(indicator.values)) {
+              indicators[1].push(indicator);
+            } else {
+              indicators[0].push(indicator);
+            }
+          });
+        }
+
+        return indicators;
+      },
+      [[], []],
+    );
+
+    // cypress' electron don't support [].flat():
+    return [...sorteIndicatorsByEmptiness[0], ...sorteIndicatorsByEmptiness[1]];
+  }
+
+  render() {
+    const { chartStatus, setDirectPathToChild } = this.props;
+    if (chartStatus === 'loading') {
+      return null;
+    }
+
+    const indicators = this.getFilterIndicators();
+    // if total indicators <= 5, show all
```

Comment:
should this be `< 5`?

Reply:
we have 5 slots for indicators. 
if total filter fields = 5, we can show all of them.
if total filter fields > 5, we should use 4 slots show each indicator, the last slot is used to show a group together.